plugins/processors/awsapplicationsignals/internal/resolver/kubernetes_utils.go

```diff
@@ -101,6 +108,40 @@ func getWorkloadAndNamespace(pod *corev1.Pod) string {
 	return workloadAndNamespace
 }
 
+// InferWorkloadName tries to parse the given podName to find the top-level workload name.
+//
+// 1) If it matches <statefulset>-<ordinal>, return <statefulset>.
+// 2) If it matches <something>-<5charSuffix>:
+//   - If <something> is <deployment>-<6–10charSuffix>, return <deployment>.
+//   - Else return <something> (likely a bare ReplicaSet or DaemonSet).
+//
+// 3) If no pattern matches, return the original podName.
+//
+// Caveat: You can't reliably distinguish DaemonSet vs. bare ReplicaSet by name alone.
+func inferWorkloadName(podName string) string {
+	// 1) Check if it's a StatefulSet pod: <stsName>-<ordinal>
+	if matches := reStatefulSet.FindStringSubmatch(podName); matches != nil {
+		return matches[1] // e.g. "mysql-0" => "mysql"
+	}
+
+	// 2) Check if it's a Pod with a 5-char random suffix: <parentName>-<5Chars>
+	if matches := replicaSetOrDaemonSetFromPodPattern.FindStringSubmatch(podName); matches != nil {
+		parentName := matches[1]
+
+		// If parentName ends with 6–10 random chars, that parent is a Deployment-based ReplicaSet.
+		// So the top-level workload is the first part before that suffix.
+		if rsMatches := deploymentFromReplicaSetPattern.FindStringSubmatch(parentName); rsMatches != nil {
+			return rsMatches[1] // e.g. "nginx-a2b3c4" => "nginx"
+		}
+
+		// Otherwise, it's a "bare" ReplicaSet or DaemonSet—just return parentName.
+		return parentName
+	}
+
+	// 3) If none of the patterns matched, return the entire podName.
+	return podName
```

Comment:
Hope this wont happen. LOL. high cardinality.

Reply:
yeah, I think the chance is low. Even when it happens, we have the metric limiter to mitigate the high cardinality issue.